aten/src/ATen/KinetoPluginAPI.cpp

```diff
@@ -0,0 +1,40 @@
+#include <c10/core/DeviceType.h>
+#include <ATen/KinetoPluginAPI.h>
+
+#include <map>
+#include <vector>
+#include <mutex>
+
+namespace at::kineto_plugin {
+
+static std::map<DeviceType, PluginFactory> plugin_factories;
+static std::mutex mutex;
```

Comment:
I am not sure if this mutex is required. If you keep it you should add it to `GetKinetoPluginRegister`. Meanwhile, std::array is a good choice, feel free to use `std::map`

Reply:
ok